Assets/Scripts/Models/Character/Health.cs

```diff
@@ -0,0 +1,179 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+
+[Serializable]
+public class Health
+{
+    //// Health States Order
+    //// Revive, Heal, Overheal, Hit, Destroyed
+
+    private float healthValue;
+
+    public Health(float maxHealth)
+    {
+        MaxHealth = maxHealth;
+        healthValue = maxHealth;
+    }
+
+    public Health(float maxHealth, bool isInvincible, bool isRevivable, bool canOveheal)
+    {
+        MaxHealth = maxHealth;
+        healthValue = maxHealth;
+        IsInvincible = isInvincible;
+        IsRevivable = isRevivable;
+        CanOverheal = canOveheal;
+    }
+
+    private Health(Health other)
+    {
+        healthValue = other.healthValue;
+        MaxHealth = other.MaxHealth;
+        IsInvincible = other.IsInvincible;
+        IsRevivable = other.IsRevivable;
+        CanOverheal = other.CanOverheal;
+    }
+
+    public event Action Revive;
+
+    public event Action Heal;
+
+    public event Action Overheal;
+
+    public event Action Hit;
+
+    public event Action Destroyed;
+
+    public float Value
+    {
+        get
+        {
+            return healthValue;
+        }
+
+        set
+        {
+            if (value > healthValue)
```

Comment:
If your talking about calling methods from the property then sure thats easy enough but if your not then are we fine with properties calling methods?